Malfunction on VATZ while make coverage, overriding a real DB  .vatz

### Checklist
- [x] Bug from Service (Vatz)
- [ ] Bug from Plugin (Vatz) 
- [ ] Malfunction on CI/CD 
- [ ] etc(e.g, Hardware version doesn't match, )

---

### **Describe the bug**
> A clear and concise description of what the bug is.

Test plugin with coverage affect on real db. 
This should be run with dummy database

### **How to Reproduce**
> Steps to reproduce the behavior:

1. Update latest vatz
```
root@burnt-testnet-validator-do-nyc1:~/dsrv# git clone https://github.com/dsrvlabs/vatz
Cloning into 'vatz'...
remote: Enumerating objects: 835, done.
remote: Counting objects: 100% (253/253), done.
remote: Compressing objects: 100% (167/167), done.
remote: Total 835 (delta 135), reused 141 (delta 85), pack-reused 582
Receiving objects: 100% (835/835), 378.06 KiB | 3.53 MiB/s, done.
Resolving deltas: 100% (406/406), done.
```

2. build vatz
```
oot@burnt-testnet-validator-do-nyc1:~/dsrv# make build
make: *** No rule to make target 'build'.  Stop.
root@burnt-testnet-validator-do-nyc1:~/dsrv# cd vatz
root@burnt-testnet-validator-do-nyc1:~/dsrv/vatz# make build
go build -ldflags="-X 'github.com/dsrvlabs/vatz/utils.Version=main' -X 'github.com/dsrvlabs/vatz/utils.Commit=ef1ae24'" -v
```

3. init vatz
```
root@burnt-testnet-validator-do-nyc1:~/dsrv/vatz# ./vatz init
2023-04-20T18:12:07Z INF init module=main
2023-04-20T18:12:07Z INF create file default.yaml module=main
2023-04-20T18:12:07Z INF initDB /root/.vatz/vatz.db module=db
2023-04-20T18:12:07Z INF newWriter /root/.vatz/vatz.db module=db
2023-04-20T18:12:07Z INF Create DB Instance module=db
```

4. Check plugin list
```
root@burnt-testnet-validator-do-nyc1:~/dsrv/vatz# ./vatz plugin list
2023-04-20T18:12:12Z INF List plugins module=plugin
2023-04-20T18:12:12Z INF List module=plugin
2023-04-20T18:12:12Z INF newReader /root/.vatz/vatz.db module=db
2023-04-20T18:12:12Z INF Create DB Instance module=db
2023-04-20T18:12:12Z INF List Plugin module=db
+------+--------------+------------+---------+
| NAME | INSTALL DATA | REPOSITORY | VERSION |
+------+--------------+------------+---------+
+------+--------------+------------+---------+
```

5. install other plugin 
```
root@burnt-testnet-validator-do-nyc1:~/dsrv/vatz# ./vatz plugin install github.com/dsrvlabs/vatz-plugin-sysutil/plugins/cpu_monitor cpu-monitor
2023-04-20T18:12:27Z INF Install new plugin github.com/dsrvlabs/vatz-plugin-sysutil/plugins/cpu_monitor at /root/.vatz module=plugin
2023-04-20T18:12:27Z INF Install new plugin github.com/dsrvlabs/vatz-plugin-sysutil/plugins/cpu_monitor module=plugin
2023-04-20T18:12:28Z INF newWriter /root/.vatz/vatz.db module=db
2023-04-20T18:12:28Z INF Create DB Instance module=db
2023-04-20T18:12:28Z INF AddPlugin module=db
```

6.  Check plugin list
```
oot@burnt-testnet-validator-do-nyc1:~/dsrv/vatz# ./vatz plugin list
2023-04-20T18:12:33Z INF List plugins module=plugin
2023-04-20T18:12:33Z INF List module=plugin
2023-04-20T18:12:33Z INF newReader /root/.vatz/vatz.db module=db
2023-04-20T18:12:33Z INF Create DB Instance module=db
2023-04-20T18:12:33Z INF List Plugin module=db
+-------------+---------------------+-------------------------------------------------------------+---------+
| NAME        | INSTALL DATA        | REPOSITORY                                                  | VERSION |
+-------------+---------------------+-------------------------------------------------------------+---------+
| cpu-monitor | 2023-04-20 18:12:28 | github.com/dsrvlabs/vatz-plugin-sysutil/plugins/cpu_monitor | latest  |
+-------------+---------------------+-------------------------------------------------------------+---------+
```

7. run `make coverage`
```
root@burnt-testnet-validator-do-nyc1:~/dsrv/vatz# make coverage
?   	github.com/dsrvlabs/vatz	[no test files]
ok  	github.com/dsrvlabs/vatz/cmd	4.404s	coverage: 19.1% of statements
ok  	github.com/dsrvlabs/vatz/manager/api	0.005s	coverage: 0.0% of statements [no tests to run]
ok  	github.com/dsrvlabs/vatz/manager/config	0.007s	coverage: 91.2% of statements
ok  	github.com/dsrvlabs/vatz/manager/dispatcher	0.005s	coverage: 7.8% of statements
ok  	github.com/dsrvlabs/vatz/manager/executor	0.006s	coverage: 68.9% of statements
ok  	github.com/dsrvlabs/vatz/manager/healthcheck	0.005s	coverage: 46.5% of statements
ok  	github.com/dsrvlabs/vatz/manager/plugin	2.590s	coverage: 68.7% of statements
?   	github.com/dsrvlabs/vatz/manager/types	[no test files]
?   	github.com/dsrvlabs/vatz/mocks	[no test files]
ok  	github.com/dsrvlabs/vatz/monitoring/prometheus	0.014s	coverage: 0.0% of statements
ok  	github.com/dsrvlabs/vatz/rpc	1.008s	coverage: 67.2% of statements
ok  	github.com/dsrvlabs/vatz/utils	0.005s	coverage: 13.3% of statements
```

8. Check plugin list

```
root@burnt-testnet-validator-do-nyc1:~/dsrv/vatz# ./vatz plugin list
2023-04-20T18:12:56Z INF List plugins module=plugin
2023-04-20T18:12:56Z INF List module=plugin
2023-04-20T18:12:56Z INF newReader /root/.vatz/vatz.db module=db
2023-04-20T18:12:56Z INF Create DB Instance module=db
2023-04-20T18:12:56Z INF List Plugin module=db
+---------------+---------------------+----------------------------------------------------------------------+---------+
| NAME          | INSTALL DATA        | REPOSITORY                                                           | VERSION |
+---------------+---------------------+----------------------------------------------------------------------+---------+
| cosmos-status | 2023-04-20 18:12:50 | github.com/dsrvlabs/vatz-plugin-cosmoshub/plugins/node_active_status | latest  |
+---------------+---------------------+----------------------------------------------------------------------+---------+
root@burnt-testnet-validator-do-nyc1:~/dsrv/vatz#
```


### **Expected behavior**
> A clear and concise description of what you expected to happen.

Plugin list remain same as before, no update on DB. 

### **Screenshots**
> If applicable, add screenshots to help explain your problem.

<img width="1182" alt="image" src="https://user-images.githubusercontent.com/6308023/233453410-03351a54-db7a-44f1-82b0-e9cc503c4f76.png">

--- a/cmd/cmd.go
+++ b/cmd/cmd.go
@@ -1,6 +1,7 @@
 package cmd
 
 import (
+	tp "github.com/dsrvlabs/vatz/manager/types"
 	"github.com/spf13/cobra"
 
 	dp "github.com/dsrvlabs/vatz/manager/dispatcher"
@@ -31,7 +32,7 @@ var (
 // CreateRootCommand creates root command of Cobra.
 func CreateRootCommand() *cobra.Command {
 	rootCmd := &cobra.Command{}
-	rootCmd.AddCommand(createInitCommand())
+	rootCmd.AddCommand(createInitCommand(tp.LIVE))
 	rootCmd.AddCommand(createStartCommand())
 	rootCmd.AddCommand(createPluginCommand())
 	rootCmd.AddCommand(createVersionCommand())

--- a/cmd/cmd_test.go
+++ b/cmd/cmd_test.go
@@ -1,6 +1,7 @@
 package cmd
 
 import (
+	tp "github.com/dsrvlabs/vatz/manager/types"
 	"os"
 	"testing"
 
@@ -28,7 +29,7 @@ func TestInitCmd(t *testing.T) {
 
 	for _, test := range tests {
 		root := cobra.Command{}
-		root.AddCommand(createInitCommand())
+		root.AddCommand(createInitCommand(tp.TEST))
 		root.SetArgs(test.Args)
 
 		err := root.Execute()

--- a/cmd/init.go
+++ b/cmd/init.go
@@ -1,6 +1,7 @@
 package cmd
 
 import (
+	tp "github.com/dsrvlabs/vatz/manager/types"
 	"os"
 
 	"github.com/rs/zerolog/log"
@@ -9,7 +10,7 @@ import (
 	"github.com/dsrvlabs/vatz/manager/plugin"
 )
 
-func createInitCommand() *cobra.Command {
+func createInitCommand(initializer tp.Initializer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "init",
 		Short: "Init",
@@ -82,7 +83,7 @@ plugins_infos:
 			}
 
 			mgr := plugin.NewManager(pluginDir)
-			return mgr.Init()
+			return mgr.Init(initializer)
 		},
 	}
 

--- a/manager/plugin/plugin.go
+++ b/manager/plugin/plugin.go
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"errors"
 	"fmt"
+	tp "github.com/dsrvlabs/vatz/manager/types"
 	"os"
 	"os/exec"
 	"strings"
@@ -30,7 +31,7 @@ type VatzPlugin struct {
 
 // VatzPluginManager provides management functions for plugin.
 type VatzPluginManager interface {
-	Init() error
+	Init(runType tp.Initializer) error
 
 	Install(repo, name, version string) error
 	List() ([]VatzPlugin, error)
@@ -45,8 +46,12 @@ type vatzPluginManager struct {
 	home string
 }
 
-func (m *vatzPluginManager) Init() error {
-	return initDB(fmt.Sprintf("%s/%s", m.home, pluginDBName))
+func (m *vatzPluginManager) Init(runType tp.Initializer) error {
+	dbName := pluginDBName
+	if runType == tp.TEST {
+		dbName = "vatz-test.db"
+	}
+	return initDB(fmt.Sprintf("%s/%s", m.home, dbName))
 }
 
 func (m *vatzPluginManager) Install(repo, name, version string) error {

--- a/manager/types/cmd.go
+++ b/manager/types/cmd.go
@@ -1,0 +1,8 @@
+package types
+
+type Initializer string
+
+const (
+	TEST Initializer = "TEST"
+	LIVE Initializer = "LIVE"
+)

